internal/docker/docker.go

```diff
@@ -301,3 +334,47 @@ func writeRequest(state *config.LibrarianState, libraryID, jsonFilePath string)
 
 	return nil
 }
+
+// readResponse reads the library state from configure-response.json.
+//
+// The response file is removed afterwards.
+func readResponse(contentLoader func(data []byte, state *config.LibraryState) error, jsonFilePath string) (*config.LibraryState, error) {
+	data, err := os.ReadFile(jsonFilePath)
+	defer func(name string) {
+		err := os.Remove(name)
+		if err != nil {
+			slog.Warn("fail to remove file", slog.String("name", name), slog.Any("err", err))
+		}
+	}(jsonFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response JSON file: %w", err)
+	}
+
+	libraryState := &config.LibraryState{}
+
+	if err := contentLoader(data, libraryState); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal JSON file, %s, to state: %w", jsonFilePath, err)
```

Comment:
This function doesn't know it's JSON, that's why it takes a contentLoader.
If you know it's JSON, remove contentLoader.

Reply:
I changed the error message.